### PR TITLE
enhance(packages/graphql): ensure that authentication is implemented correctly

### PR DESCRIPTION
**Pull request by sjschlapbach**
(more detailed description -> see comments)

### CODE REVIEW

File: packages/graphql/src/schema/query.ts [hunk 1]
```diff
@@ -261,7 +284,8 @@ export const Query = builder.queryType({
           return SessionService.getPinnedFeedbacks(args, ctx)
         },
       }),
-      course: asAuthenticated.field({
+
+      course: asUser.field({
```
Comment (sjschlapbach) [inline packages/graphql/src/schema/query.ts:288]: Changed `asAuthenticated` to `asUser` as this query is only used in `frontend-manage`

File: packages/graphql/src/schema/query.ts [hunk 2]
```diff
@@ -271,7 +295,8 @@ export const Query = builder.queryType({
           return CourseService.getCourseData(args, ctx)
         },
       }),
-      liveSession: t.field({
+
+      liveSession: asUser.field({
```
Comment (sjschlapbach) [inline packages/graphql/src/schema/query.ts:299]: Changed `asAuthenticated` to `asUser` as this query is only used in `frontend-manage` and `frontend-control`

File: packages/graphql/src/schema/query.ts [hunk 3]
```diff
@@ -291,7 +317,8 @@ export const Query = builder.queryType({
           return QuestionService.getSingleQuestion(args, ctx)
         },
       }),
-      singleMicroSession: asAuthenticated.field({
+
+      singleMicroSession: asUser.field({
```
Comment (sjschlapbach) [inline packages/graphql/src/schema/query.ts:321]: Changed from `asAuthenticated` to `asUser` as this query is only used in `frontend-manage`

File: packages/graphql/src/schema/query.ts [hunk 4]
```diff
@@ -331,7 +361,8 @@ export const Query = builder.queryType({
           return ParticipantService.getParticipations(args, ctx)
         },
       }),
-      getCourseOverviewData: asAuthenticated.field({
+
+      getCourseOverviewData: asParticipant.field({
```
Comment (sjschlapbach) [inline packages/graphql/src/schema/query.ts:365]: Changed `asAuthenticated` to `asParticipant` as this course query is only used in `frontend-pwa`

File: packages/graphql/src/schema/query.ts [hunk 5]
```diff
@@ -353,7 +385,8 @@ export const Query = builder.queryType({
           return ParticipantGroupService.getGroupActivityDetails(args, ctx)
         },
       }),
-      getBookmarkedQuestions: t.field({
+
+      getBookmarkedQuestions: asParticipant.field({
```
Comment (sjschlapbach) [inline packages/graphql/src/schema/query.ts:389]: Changed from no authentication at all to `asParticipant` as this query is only used in `frontend-pwa`

**Review by sjschlapbach — COMMENTED** [review]
A few access rights have been changed with corresponding inline comments.
For the `participantGroups`, it could also be considered to change the access rights from `asAuthenticated` to `asParticipant` as it is currently only used on `frontend-pwa`. However, it will probably also be used on the lecturer view at some point?

File: packages/graphql/src/services/feedbacks.ts [hunk 6]
```diff
@@ -98,7 +98,7 @@ export async function createFeedback(
 // add response to an existing feedback
 export async function respondToFeedback(
   { id, responseContent }: { id: number; responseContent: string },
-  ctx: Context
+  ctx: ContextWithUser
 ) {
   const feedback = await ctx.prisma.feedback.update({
```
Comment (rschlaefli) [inline packages/graphql/src/services/feedbacks.ts]: verify with include and where using ctx.user.sub

File: packages/graphql/src/services/participants.ts [hunk 7]
```diff
@@ -352,14 +347,14 @@ interface BookmarkQuestionArgs {
 
 export async function bookmarkQuestion(
   { instanceId, courseId, bookmarked }: BookmarkQuestionArgs,
-  ctx: Context
+  ctx: ContextWithUser
 ) {
   console.log('received args', { instanceId, courseId, bookmarked })
```
Comment (rschlaefli) [inline packages/graphql/src/services/participants.ts]: remove console.log

File: packages/graphql/src/schema/query.ts (hunk 5 above)
Comment (sjschlapbach) [reply packages/graphql/src/schema/query.ts:389]: TODO: ensure that the access rights here are consistent with the new working implementation on the stacked learning element branch

Comment (sjschlapbach) [reply packages/graphql/src/schema/query.ts:389]: Permission settings from the stacked learning element branch were not overwritten during merge, so they should be correct as tested there.

**Review by sjschlapbach — COMMENTED** [review]
@rschlaefli Now after we merged the new stacked learning elements, I checked the permissions after the merge, and they did not change. So this branch should also be ready to be merged

**Review by rschlaefli — APPROVED** [review]

**Pull request merged**